From 4b0744ae296c45693d035ccce26a5ad17bd48df6 Mon Sep 17 00:00:00 2001
From: Fugang Duan <fugang.duan@nxp.com>
Date: Wed, 17 Jan 2018 15:40:32 +0800
Subject: [PATCH 3224/5242] MLK-17400-06 arm64: dts: 8qxp: add enet2 phy
 regulator for arm2 board

commit  d5bd92f11dd240943e4fa8d68bbad8d5690ee691 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add enet2 phy regulator for arm2 board to set IO voltage to 1.8v.

Signed-off-by: Fugang Duan <fugang.duan@nxp.com>
Acked-by: Gao Pan <pandy.gao@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2.dts |   17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2.dts
index 319ac32..2a7702a 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2.dts
@@ -59,6 +59,15 @@
 			gpio = <&gpio5 9 GPIO_ACTIVE_HIGH>;
 			enable-active-high;
 		};
+
+		reg_fec2_supply: fec2_nvcc {
+			compatible = "regulator-fixed";
+			regulator-name = "fec2_nvcc";
+			regulator-min-microvolt = <1800000>;
+			regulator-max-microvolt = <1800000>;
+			gpio = <&max7322 0 GPIO_ACTIVE_HIGH>;
+			enable-active-high;
+		};
 	};
 
 	sound-cs42888 {
@@ -424,6 +433,7 @@
 	pinctrl-0 = <&pinctrl_fec2>;
 	phy-mode = "rgmii-id";
 	phy-handle = <&ethphy1>;
+	phy-supply = <&reg_fec2_supply>;
 	fsl,magic-packet;
 	status = "disabled";
 };
@@ -514,6 +524,13 @@
 	pinctrl-0 = <&pinctrl_lpi2c1>;
 	status = "okay";
 
+	max7322: gpio@68 {
+		compatible = "maxim,max7322";
+		reg = <0x68>;
+		gpio-controller;
+		#gpio-cells = <2>;
+	};
+
 	typec_ptn5150: typec@3d {
 		compatible = "nxp,ptn5150";
 		pinctrl-names = "default";
-- 
1.7.9.5

